java/arcs/core/data/Ttl.kt

```diff
@@ -24,15 +26,33 @@ sealed class Ttl(count: Int, val isInfinite: Boolean = false) {
             "must be either positive count on infinite, " +
                 "but got count=$count and isInfinite=$isInfinite"
         }
+        if (!isInfinite) {
+            requireNotNull(time) {
+                "time cannot be null for non infinite ttl"
+            }
+        }
     }
 
     override fun equals(other: Any?): Boolean =
         other is Ttl && minutes == other.minutes
 
     override fun hashCode(): Int = minutes.hashCode()
 
-    data class Minutes(val count: Int) : Ttl(count)
-    data class Hours(val count: Int) : Ttl(count)
-    data class Days(val count: Int) : Ttl(count)
-    object Infinite : Ttl(-1, true)
+    @Suppress("")
+    fun calculateExpiration(): Long {
+        if (isInfinite) {
+            return RawEntity.NO_EXPIRATION
+        } else {
+            return requireNotNull(time).currentTimeMillis.plus(minutes * 60 * 1000)
+        }
+    }
+
+    data class Minutes(val count: Int) : Ttl(Ttl.time, count)
+    data class Hours(val count: Int) : Ttl(Ttl.time, count)
+    data class Days(val count: Int) : Ttl(Ttl.time, count)
+    object Infinite : Ttl(null, -1, true)
+
+    companion object {
+        var time: Time? = null
```

Comment:
this seems potentially dangerous having a global variable like this

Reply:
need your advice on how to best avoid it...